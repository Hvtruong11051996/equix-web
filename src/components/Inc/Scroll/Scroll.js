import scrollbarWidth from 'scrollbarwidth';

export default class Scroll {
  constructor(scroll) {
    if (!scroll || scroll.hasAttribute('scroll-content')) return;
    scroll.setAttribute('scroll-content', '');
    scroll.style.bottom = scroll.style.right = (0 - scrollbarWidth()) + 'px';
    const container = scroll.parentNode;
    container.setAttribute('scroll', '');
    let div;
    let bar1;
    scroll.childNodes.forEach(node => {
      if (node.classList && node.classList.contains('vertical')) {
        bar1 = node.firstChild
      }
    })
    if (!bar1) {
      div = document.createElement('div');
      div.className = 'vertical';
      div.style.display = 'none';
      bar1 = document.createElement('div');
      bar1.className = 'scrollBar';
      div.appendChild(bar1);
      container.appendChild(div);
    }
    bar1.style.height = (scroll.clientHeight / (scroll.scrollHeight / scroll.clientHeight)) + 'px';
    bar1.onmousedown = function(e) {
      // bar1.dragging = true;
      e = e || window.event;
      // get the mouse cursor position at startup:
      bar1.topPos = e.clientY - bar1.offsetTop;
      document.body.classList.add('noSelect');
      document.onmouseup = function() {
        /* stop moving when mouse button is released: */
        // bar1.dragging = false;
        document.onmouseup = null;
        document.onmousemove = null;
        document.body.classList.remove('noSelect');
      };
      // call a function whenever the cursor moves:
      document.onmousemove = function(e) {
        e = e || window.event;
        let top = e.clientY - bar1.topPos;

        if (top < 0) top = 0;
        else if (top > container.clientHeight - bar1.clientHeight) top = container.clientHeight - bar1.clientHeight;
        // bar1.style.top = top + "px";
        scroll.scrollTop = top * (scroll.scrollHeight - scroll.clientHeight) / (bar1.parentNode.clientHeight - bar1.clientHeight);
        // const elmnt = e.target.parentNode.parentNode;
        // elmnt.style.top = (elmnt.offsetTop - e.target.pos2) + "px";
        // elmnt.style.left = (elmnt.offsetLeft - e.target.pos1) + "px";
      };
    };
    let bar2;
    scroll.childNodes.forEach(node => {
      if (node.classList && node.classList.contains('horizontal')) {
        bar2 = node.firstChild
      }
    })
    if (!bar2) {
      div = document.createElement('div');
      div.className = 'horizontal';
      div.style.display = 'none';
      bar2 = document.createElement('div');
      bar2.className = 'scrollBar';
      div.appendChild(bar2);
      container.appendChild(div);
    }
    bar2.style.width = (scroll.clientWidth / (scroll.scrollWidth / scroll.clientWidth)) + 'px';
    bar2.onmousedown = function(e) {
      // bar1.dragging = true;
      e = e || window.event;
      // get the mouse cursor position at startup:
      bar2.leftPos = e.clientX - bar2.offsetLeft; //
      document.body.classList.add('noSelect');
      document.onmouseup = function() {
        /* stop moving when mouse button is released: */
        // bar1.dragging = false;
        document.onmouseup = null;
        document.onmousemove = null;
        document.body.classList.remove('noSelect');
      };
      // call a function whenever the cursor moves:
      document.onmousemove = function(e) {
        e = e || window.event;
        let left = e.clientX - bar2.leftPos;

        if (left < 0) left = 0;
        else if (left > container.clientWidth - bar2.clientWidth) left = container.clientWidth - bar2.clientWidth;
        scroll.scrollLeft = left * (scroll.scrollWidth - scroll.clientWidth) / (bar2.parentNode.clientWidth - bar2.clientWidth);
      };
    };

    scroll.onscroll = function() {
      if (scroll.onmousemove) scroll.onmousemove();
      bar1.style.transform = `translate(0,${(scroll.scrollTop * (bar1.parentNode.clientHeight - bar1.clientHeight) / (scroll.scrollHeight - scroll.clientHeight))}px)`;
      bar2.style.transform = `translate(${(scroll.scrollLeft * (bar2.parentNode.clientWidth - bar2.clientWidth) / (scroll.scrollWidth - scroll.clientWidth))}px,0)`;
    };
    bar1.parentNode.onmouseenter = bar2.parentNode.onmouseenter = function() {
      scroll.onmousemove = null;
      if (scroll.time) {
        clearTimeout(scroll.time);
      }
      container.setAttribute('active', '');
    };
    bar1.parentNode.onmouseleave = bar2.parentNode.onmouseleave = function() {
      scroll.onmousemove = scroll.timmeout;
      container.removeAttribute('active');
    };
    scroll.timmeout = function() {
      container.setAttribute('active', '');
      if (this.time) {
        clearTimeout(this.time);
      }
      this.time = setTimeout(function() {
        container.removeAttribute('active');
        this.time = null;
      }, 1000);
    };
    scroll.onmousemove = scroll.timmeout;
    scroll.monitor = () => {
      let size = 0;
      if (scroll.lastChild) {
        size = scroll.lastChild.offsetHeight + scroll.lastChild.offsetTop + 1;
        if (scroll.lastElementChild) size += parseInt(getComputedStyle(scroll.lastElementChild).marginBottom);
      }
      if (scroll.parentNode) {
        scroll.parentNode.style.height = size + 'px';
      }
      if (bar1.parentNode) {
        if (scroll.scrollHeight <= scroll.clientHeight) {
          bar1.parentNode.style.display = 'none';
        } else {
          bar1.parentNode.style.display = '';
          size = (scroll.clientHeight / (scroll.scrollHeight / scroll.clientHeight));
          bar1.style.height = (size < 32 ? 32 : size) + 'px';
          bar1.style.transform = `translate(0,${(scroll.scrollTop * (bar1.parentNode.clientHeight - bar1.clientHeight) / (scroll.scrollHeight - scroll.clientHeight))}px)`;
        }
      }
      if (bar2.parentNode) {
        if (scroll.scrollWidth === scroll.clientWidth) {
          bar2.parentNode.style.display = 'none';
        } else {
          bar2.parentNode.style.display = '';
          size = (scroll.clientWidth / (scroll.scrollWidth / scroll.clientWidth));
          bar2.style.width = (size < 32 ? 32 : size) + 'px';
          bar2.style.transform = `translate(${(scroll.scrollLeft * (bar2.parentNode.clientWidth - bar2.clientWidth) / (scroll.scrollWidth - scroll.clientWidth))}px,0)`;
        }
      }
    };
    scroll.addEventListener('animationstart', (event) => {
      if (event && event.animationName !== 'nodeVisible') return;
      const scrollContent = scroll.parentNode.querySelector('[scroll-content]')
      if (scrollContent && scroll.parentNode.classList.contains('ag-body-viewport-wrapper')) {
        const headerContainer = scroll.parentNode.parentNode.parentNode.querySelector('.ag-header-container')
        if (headerContainer) {
          headerContainer.style.left = '0px'
        }
      }
      scroll.monitor()
    }, false);
    container.ro = new MutationObserver(scroll.monitor);
    scroll.monitor();
    setTimeout(() => {
      scroll.monitor();
    }, 500)
    container.ro.observe(scroll, {
      attributes: true,
      childList: true,
      subtree: true
    });
  }
}
