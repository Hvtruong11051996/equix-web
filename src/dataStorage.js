
import requireTime from './constants/require_time';
import config from '../public/config'

const obj = {
    mode: null,
    timeoutNetwork: null,
    isChangePassword: false,
    closeChartLayout: {},
    isSessionAnotherLogin: false,
    dicClearRealtime: {},
    materialFont: null,
    listMapping: [],
    deviceLang: 'en_US',
    onChangeLang: null,
    dicAccounts: {},
    reloadNews: [],
    downloadFunc: {},
    printFunc: {},
    layout_role: null,
    force_update_create_user: false,
    deletingLayout: '',
    deletingChartLayout: '',
    isDragChart: false,
    isDefaultLayout: true,
    chartSelfChange: false,
    logout: null,
    isChangeLayout: false,
    curLayoutVersion: null,
    listLayout: {},
    usingLayout: '',
    curChartLayoutVersion: '1',
    listChartLayout: {},
    usingChartLayout: '',
    listUsingChartLayout: {},
    renderAfterRealtime: null,
    checkVersionLayoutBeforeOverride: null,
    isLogin: false,
    uid: null,
    isGuest: true,
    connected: true,
    full_name: null,
    accessToken: null,
    pin: null,
    user_details: null,
    user_type: 2,
    logId: 0,
    timeInterval: 0,
    isIntervalRealtime: 0,
    deviceId: 0,
    account_id: null,
    accountInfo: null,
    role: null,
    sseRegisters: {},
    handleClickOutside: null,
    typeRegisters: {},
    loginEmail: null,
    accountStreamingUrl: null,
    notify: null,
    userInfo: null,
    accounts: [],
    subPath: '',
    checkUpdate: false,
    checkLoginPin: null,
    intervalRefreshToken: null,
    intervalRefreshTokenGuest: null,
    showNotification: null,
    isShowNotification: true,
    showCancelled: false,
    showExpired: false,
    showFilled: false,
    showOnMarket: false,
    showRejected: false,
    showPartialFill: false,
    showAllNews: true,
    translate: null,
    lang: '',
    dicPersonal: {},
    dicPosition: {},
    dicNotiShowed: {},
    symbolEquity: {},
    clickedMenu: false,
    showPin: false,
    resolution: null,
    dicAccount: {},
    accountsObjDic: {},
    symbolsObjDic: {},
    pinWasChanged: false,
    symbolHistoryInfo: {},
    checkPinCallback: null,
    requireTime: requireTime.ON_CHANGE,
    verifiedPin: false,
    isStayLogin: false,
    getLayout: null,
    listFilledCancelled: {},
    dicInterval: {},
    dicLevel1: {},
    dicDepth: {},
    dicCos: {},
    dicLevel1RealTime: {},
    dicDepthRealTime: {},
    dicCosRealTime: {},
    global_symbol: null,
    listUserWatchlist: {},
    usingWatchlist: '',
    numberNoticeNews: null,
    lastSymbolSearchNews: null,
    removeSaveChartLayout: null,
    usingWatchlist_id: null,
    usingWatchlist_name: null,
    userGroupArray: [],
    userGroupDic: [], // tuanphan use
    dicEmailTemp: [],
    apiAccessDic: [],
    userGroupDicReverse: {},
    dicRole: {},
    marketData: {},
    allowDisableModifyCancelOrder: false,
    callBackReloadTheme: {},
    hrefImg: '',
    color: '#fff',
    linkColor: null,
    lstAccountDropdown: null,
    allowCloseAllWidgets: false,
    timeZone: '',
    symbolDefault: null,
    setVisibility: {},
    dicbyDisplayName: {},
    dicsymbol: {},
    watchlist: [],
    openNameWidget: '',
    symbolObjDepth: {},
    dictAccountNoti: [],
    beforeLogin: false,
    timeHoliday: {},
    isHolidayAU: {},
    isHolidayUS: {},
    timeServer: '',
    currentEnv: '',
    env_config: null,
    web_config: null,
    guest_config: null,
    isWarning: {
        isNotiTechnical: false,
        isNotiCashNegative: false
    },
    warningFunc: null,
    listFunctionInit: [],
    marketDataType: {},
    openingAccount: {},
    branchObjDic: {},
    session: {}
}

const objProxy = new Proxy(obj, {
    set: function (target, key, value) {
        // console.log(`YOLO ${key} ${target[key]} =====> ${value}`)
        target[key] = value;
        return true;
    }
})

export default objProxy
