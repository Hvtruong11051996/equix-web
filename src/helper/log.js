import { exception } from 'react-ga';
import { sendLogToServer } from './functionUtils'
const info = (...arg) => {
    console.info(...arg);
}
const error = (...arg) => {
    console.error(...arg);
}
const warn = (...arg) => {
    sendLogToServer(...arg);
    console.warn(...arg);
}
const log = (...arg) => {
    console.log(...arg);
    sendLogToServer(...arg);
}

const sendLog = (...arg) => {
    sendLogToServer(...arg);
}
const sendLogError = (...arg) => {
    sendLogToServer(...arg, 'error');
}

export default {
    info,
    error,
    warn,
    log,
    sendLogError,
    sendLog
}
