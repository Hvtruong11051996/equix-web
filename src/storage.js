export const storage = {
    emitter: {}
}

export const func = {
    setStore: (name, emitter) => {
        storage.emitter[name] = emitter;
    },
    getStore: (name) => {
        return storage.emitter[name];
    },
    emitter: (name, eventName, data) => {
        const emit = func.getStore(name);
        emit && emit.emit(eventName, data);
    }
}
