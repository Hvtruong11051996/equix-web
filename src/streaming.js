
import { getAccountStreamingUrl, getUserStreamingUrl, getStreamingAccountUrl, getAllAccountsStreamingUrl, getBranchStreamingUrl, getWarningTechnicalStreamingUrl } from './helper/request';
import { subcriber, unregister } from './sse';
import { addPopoutEventHubListener, isSubWindow } from './helper/functionUtils'
import logger from './helper/log';
import dataStorage from './dataStorage';
import config from '../public/config';

export function regisRealtime(obj = {}) {
    if (!obj.url) return
    if (isSubWindow()) {
        addPopoutEventHubListener(obj)
        return
    }
    subcriber(obj.url, obj.callback, obj.type);
}

export function unregisRealtime(obj = {}) {
    if (!obj.url) return
    if (isSubWindow()) {
        removePopoutEventHubListener(obj)
        return
    }
    unregister(obj.url, obj.callback, obj.type);
}

export function registerAccount(accountId, callback, type) {
    const url = accountId ? dataStorage.accountStreamingUrl : null;
    regisRealtime({ url, callback, type })
}
export function unregisterAccount(accountId, callback, type) {
    const url = accountId ? dataStorage.accountStreamingUrl : null;
    unregisRealtime({ url, callback, type })
}
export function registerAccountStreaming(accountId, callback, type, isOperator = false) {
    const url = accountId ? isOperator ? getAllAccountsStreamingUrl() : getStreamingAccountUrl(accountId) : null;
    dataStorage.accountStreamingUrl = url;
    regisRealtime({ url, callback, type })
}

export function registerAllOrders(callback, type) {
    const url = dataStorage.accountStreamingUrl
    regisRealtime({ url, callback, type })
}

export function unregisterAllOrders(callback, type) {
    const url = dataStorage.accountStreamingUrl;
    unregisRealtime({ url, callback, type })
}

export function registerUser(userId, callback, type) {
    const url = userId ? getUserStreamingUrl(userId) : null;
    regisRealtime({ url, callback, type })
}
export function unregisterUser(userId, callback, type) {
    const url = userId ? getUserStreamingUrl(userId, userId === 'guest' ? dataStorage.web_config['guest'] : null) : null;
    unregisRealtime({ url, callback, type })
}
export function registerBranch(userId, callback, type) {
    const url = userId ? getAllAccountsStreamingUrl(userId) : null;
    regisRealtime({ url, callback, type })
}
export function unregisterBranch(userId, callback, type) {
    const url = userId ? getAllAccountsStreamingUrl(userId) : null;
    unregisRealtime({ url, callback, type })
}
export function registerWarningTechnicalStreaming(userId, callback, type) {
    const url = userId ? getWarningTechnicalStreamingUrl() : null;
    regisRealtime({ url, callback, type })
}
export function unregisterWarningTechnicalStreaming(userId, callback, type) {
    const url = userId ? getWarningTechnicalStreamingUrl() : null;
    unregisRealtime({ url, callback, type })
}
