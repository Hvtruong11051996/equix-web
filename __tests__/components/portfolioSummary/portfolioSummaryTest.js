import Compoment from '../../../src/components/PortfolioSummary/PortfolioSummary';
import * as request from '../../../src/helper/request';
import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react'
import dataStorage from '../../../src/dataStorage'
import log from '../../../src/helper/log';
import * as functionUtils from '../../../src/helper/functionUtils';
import Scroll from '../../../src/components/Inc/Scroll/Scroll';
import * as storage from '../../../src/storage';
import * as streamingFunc from '../../../src/streaming';
import styles from '../../../src/components/PortfolioSummary/PortfolioSummary.module.css'
import showModal from '../../../src/components/Inc/Modal/Modal';

global.MutationObserver = class {
    disconnect() { }
    observe(element, initObject) { }
};

global.Worker = class {
    postMessage() { }
    addEventListener() { }
    disconnect() { }
    observe(element, initObject) { }
};

global.URL.createObjectURL = jest.fn();
let testfunc = jest.spyOn(streamingFunc, 'unregisterOrders')
const props1 = {
    loadState: function() {
        return {

        }
    },
    glContainer: {
        on: function(key, cb) {
            jest.spyOn(functionUtils, 'hideElement')
                .mockImplementation(() => { })
            cb()
        },
        _contentElement: [],
        _element: [{
            contains: function() {
                return true
            },
            react: function() {
            },
            removeChild: function() {
                return true
            }
        }],
        isHidden: true
    },
    confirmClose: function() {
    },
    loading: function() {
    },
    resize: function(cb) {
        cb()
    },
    receive: function() {
    },
    send: function() {
    },
    saveState: function() {
    },
    hideElement: function() {
    }
}

dataStorage.translate = () => { }
let x;
beforeEach(() => {
    x = new Compoment(props1);
});

test('portfolioSummary', () => {
    storage.func = {
        setStore: (name, emitter) => {
            return { addListener: () => { } };
        },
        getStore: (name) => {
            return { addListener: () => { } };
        },
        emitter: (name, eventName, data) => {
            return { addListener: () => { } };
        }
    }
    let test = render(<Compoment {...props1} />);
})
test('render', () => {
    x.state.accountObj.account_id = 1;
    x.render();
    expect(x.render).toMatchSnapshot();
})
test('render', () => {
    x.state = null;
    let logger = jest.spyOn(log, 'error').mockImplementation(() => { });
    x.render();
    expect(logger).toBeCalled();
})
test('componentDidMount', () => {
    x.subscription = null;
    let logger = jest.spyOn(log, 'error').mockImplementation(() => { });
    x.componentDidMount();
    expect(logger).toBeCalled();
})
test('changeAccount', () => {
    let data = true;
    jest.spyOn(functionUtils, 'getSymbolAccountWhenFirstOpenLayout')
        .mockImplementation(() => {
            return { newAccount: { account_id: 1 } };
        })
    x.setState = jest.fn(({ }, cb) => {
        if (typeof cb === 'function') cb()
    });
    x.state = { accountObj: { account_id: 1 } }

    x.changeAccount(data);
    expect(testfunc).not.toBeCalled();
})

test('componentWillUnmount', () => {
    x.emitID = { remove: () => { } }
    x.emitConnectionID = { remove: () => { } }
    x.emitRefreshID = { remove: () => { } }
    x.state.accountObj.account_id = 1;
    x.componentWillUnmount()
    expect(testfunc).toBeCalled();
})
test('componentWillUnmount', () => {
    x.state = null
    let logger = jest.spyOn(log, 'error').mockImplementation(() => { });
    x.componentWillUnmount()
    expect(logger).toBeCalled();
})

test('changeAccount', () => {
    let data = false;
    jest.spyOn(functionUtils, 'getSymbolAccountWhenFirstOpenLayout')
        .mockImplementation(() => {
            return { newAccount: {} };
        })
    x.setState = jest.fn(({ }, cb) => {
        if (typeof cb === 'function') cb()
    });
    x.changeAccount(data);
    expect(x.setState).not.toBeCalled();
})
test('changeAccount', () => {
    let data = true;
    jest.spyOn(functionUtils, 'getSymbolAccountWhenFirstOpenLayout')
        .mockImplementation(() => {
            return { newAccount: { account_id: 1 } };
        })
    x.setState = jest.fn(({ }, cb) => {
        if (typeof cb === 'function') cb()
    });
    x.changeAccount(data);
    expect(x.setState).toBeCalled();
})

test('shouldComponentUpdate', () => {
    let logger = jest.spyOn(log, 'error').mockImplementation(() => { });
    x.shouldComponentUpdate();
    expect(logger).toBeCalled()
})

test('shouldComponentUpdate', () => {
    let nextState = { isHidden: true };
    expect(x.shouldComponentUpdate({}, nextState)).toEqual(false);
})

test('shouldComponentUpdate', () => {
    let nextState = { isHidden: false };
    expect(x.shouldComponentUpdate({}, nextState)).toEqual(true);
})
test('shouldComponentUpdate', () => {
    let nextState = { isHidden: false };
    dataStorage.checkUpdate = true;
    let checkPropsStateShouldUpdate = jest.spyOn(functionUtils, 'checkPropsStateShouldUpdate')
        .mockImplementation(() => { })
    x.shouldComponentUpdate({}, nextState)
    expect(checkPropsStateShouldUpdate).toBeCalled();
})
test('shouldComponentUpdate', () => {
    let nextState = null;
    let logger = jest.spyOn(log, 'error').mockImplementation(() => { });
    x.shouldComponentUpdate({}, nextState)
    expect(logger).toBeCalled();
})
test('refreshData', () => {
    x.state.data = { a: 1, b: 2, c: 3 };
    const logger = jest.spyOn(log, 'log');
    x.setState = jest.fn(({ }, cb) => {
        if (typeof cb === 'function') cb()
    });
    x.getDataAccount = jest.fn();
    x.refreshData('refresh');
    expect(x.getDataAccount).toBeCalled();
})

test('refreshData', () => {
    x.state.data = { a: 1, b: 2, c: 3 };
    const logger = jest.spyOn(log, 'log');
    x.setState = jest.fn(({ }, cb) => {
        if (typeof cb === 'function') cb()
    });
    x.getDataAccount = jest.fn();
    x.refreshData('reabcfresh');
    expect(x.getDataAccount).not.toBeCalled();
})

test('refreshData', () => {
    x.state = null;
    const logger = jest.spyOn(log, 'error');
    x.refreshData('refresh');
    expect(logger).toBeCalled();
})

test('mappingObject', () => {
    x.setState = jest.fn()
    x.mappingObject();
    expect(x.setState).toBeCalled();
})

test('realtimeDataBalances', () => {
    x.setState = jest.fn()
    x.state.accountObj.account_id = 1
    x.realtimeDataBalances({ account_id: 1 });
    expect(x.setState).toBeCalled();
})

test('realtimeDataBalances', () => {
    x.setState = jest.fn()
    x.state.accountObj.account_id = 2
    x.realtimeDataBalances({ account_id: 1 });
    expect(x.setState).not.toBeCalled();
})

test('dataReceivedFromSearchAccount', () => {
    x.changeAccount = jest.fn()
    x.dataReceivedFromSearchAccount(9999);
    expect(x.changeAccount).toBeCalled();
})

test('dataReceivedFromSearchAccount', () => {
    x.changeAccount = jest.fn()
    x.dataReceivedFromSearchAccount(null);
    expect(x.changeAccount).not.toBeCalled();
})

test('getCashByAccount full complete', () => {
    let getData = jest.spyOn(request, 'getData')
        .mockImplementation(() => new Promise(resolve => {
            resolve({ data: { available_balance: 1 } })
        }))
    x.account_id = 1;
    x.setState = jest.fn()
    x.getCashByAccount();
    setTimeout(() => {
        expect(x.setState).toHaveBeenCalled();
    }, 0);
})

test('getCashByAccount getData get null  data', () => {
    let getData = jest.spyOn(request, 'getData')
        .mockImplementation(() => new Promise(resolve => {
            resolve({ data: null })
        }))
    x.account_id = 1;
    x.setState = jest.fn()
    x.getCashByAccount();
    setTimeout(() => {
        expect(x.setState).toHaveBeenCalled();
    }, 0);
})

test('getCashByAccount getData not work', () => {
    let logger = jest.spyOn(log, 'log').mockImplementation(() => { });
    let getData = jest.spyOn(request, 'getData')
        .mockImplementation(() => new Promise(() => {
            reject('error')
        }))
    x.account_id = 1;
    x.setState = jest.fn()
    x.getCashByAccount();
    setTimeout(() => {
        expect(logger).toHaveBeenCalled();
    }, 0);
})

test('getCashByAccount get error before getData', () => {
    let logger = jest.spyOn(log, 'error').mockImplementation(() => { });
    x.props.loading = null;
    x.account_id = 1;
    x.getCashByAccount();
    expect(logger).toHaveBeenCalled();
})

test('getCashByAccount no account_id', () => {
    x.account_id = 0;
    props1.loading = jest.fn();
    x.getCashByAccount();
    expect(props1.loading).not.toBeCalled();
})

test('tempFormatDigit', () => {
    x.state.currency = 'USD';
    x.tempFormatDigit(123, 'red');
    expect(x.tempFormatDigit).toMatchSnapshot();
})

test('tempFormatDigit', () => {
    x.tempFormatDigit('123', 'red');
    expect(x.tempFormatDigit).toMatchSnapshot();
})

test('tempFormatDigit', () => {
    x.tempFormatDigit('sad', '');
    expect(x.tempFormatDigit).toMatchSnapshot();
})
test('tempFormatDigit', () => {
    x.tempFormatDigit('');
    expect(x.tempFormatDigit).toMatchSnapshot();
})

test('tempFormatDigitPercent', () => {
    x.tempFormatDigitPercent(123);
    expect(x.tempFormatDigitPercent).toMatchSnapshot();
})

test('tempFormatDigitPercent', () => {
    x.tempFormatDigitPercent('123');
    expect(x.tempFormatDigitPercent).toMatchSnapshot();
})
test('tempFormatDigitPercent', () => {
    x.tempFormatDigitPercent('--');
    expect(x.tempFormatDigitPercent).toMatchSnapshot();
})
test('tempFormatDigitPercent', () => {
    x.tempFormatDigitPercent(-123);
    expect(x.tempFormatDigitPercent).toMatchSnapshot();
})
test('tempFormatDigitPercent', () => {
    x.tempFormatDigitPercent(0);
    expect(x.tempFormatDigitPercent).toMatchSnapshot();
})

test('tempFormatDigitPercent', () => {
    x.tempFormatDigitPercent('----');
    expect(x.tempFormatDigitPercent).toMatchSnapshot();
})

test('tempFormatDigitColor', () => {
    x.tempFormatDigitColor(123, true);
    expect(x.tempFormatDigitColor).toMatchSnapshot();
})
test('tempFormatDigitColor', () => {
    x.tempFormatDigitColor(0, true);
    expect(x.tempFormatDigitColor).toMatchSnapshot();
})
test('tempFormatDigitColor', () => {
    x.tempFormatDigitColor('0', true);
    expect(x.tempFormatDigitColor).toMatchSnapshot();
})
test('tempFormatDigitColor', () => {
    x.tempFormatDigitColor(-12, true);
    expect(x.tempFormatDigitColor).toMatchSnapshot();
})

test('tempFormatDigitColor', () => {
    x.tempFormatDigitColor(-12, false);
    expect(x.tempFormatDigitColor).toMatchSnapshot();
})

test('tempFormatDigitColor', () => {
    x.tempFormatDigitColor('--', true);
    expect(x.tempFormatDigitColor).toMatchSnapshot();
})

test('tempFormatDigitColor', () => {
    x.tempFormatDigitColor('--', false);
    expect(x.tempFormatDigitColor).toMatchSnapshot();
})

test('tempFormatDigitColor', () => {
    x.tempFormatDigitColor('---', true);
    expect(x.tempFormatDigitColor).toMatchSnapshot();
})

test('tempFormatDigitColor', () => {
    x.tempFormatDigitColor('0', false);
    expect(x.tempFormatDigitColor).toMatchSnapshot();
})

test('tempFormatDigitColor', () => {
    x.state.currency = 'USD'
    x.tempFormatDigitColor(12, true);
    expect(x.tempFormatDigitColor).toMatchSnapshot();
})
test('tempFormatDigitColor', () => {
    x.state.currency = 'USD'
    x.tempFormatDigitColor(12, false);
    expect(x.tempFormatDigitColor).toMatchSnapshot();
})

test('addClass', () => {
    x.setState = jest.fn();
    x.addClass();
    expect(x.setState).toBeCalled();
})

test('handleClickIcon', () => {
    x.handleClickIcon();
    expect(showModal).toMatchSnapshot();
})

test('renderInfoInline', () => {
    x.renderInfoInline();
    expect(x.renderInfoInline).toMatchSnapshot();
})
test('renderInfoInline', () => {
    let valueFunc = () => {
        return true
    }
    let valueFuncRight = () => {
        return true
    }
    x.renderInfoInline('', valueFunc, valueFuncRight, true, true, true, false);
    expect(x.renderInfoInline).toMatchSnapshot();
})
test('renderInfoInline', () => {
    let valueFunc = () => {
        return true
    }
    let valueFuncRight = () => {
        return true
    }
    x.renderInfoInline('', valueFunc, valueFuncRight, true, true, true, true);
    expect(x.renderInfoInline).toMatchSnapshot();
})

test('renderInfoToggle', () => {
    x.renderInfoToggle();
    expect(x.renderInfoToggle).toMatchSnapshot();
})

test('renderInfoTxt', () => {
    x.renderInfoTxt();
    expect(x.renderInfoTxt).toMatchSnapshot();
})

test('renderInfoTxt', () => {
    x.state.show = true;
    x.renderInfoTxt();
    expect(x.renderInfoTxt).toMatchSnapshot();
})

test('getDataAccount', () => {
    x.state.accountObj = { account_id: 1 };
    let logger = jest.spyOn(log, 'error').mockImplementation(() => { })
    let geturl = jest.spyOn(request, 'getUrlAnAccount')
        .mockImplementation(() => {
            return 'tren'
        })
    let getUrlTotal = jest.spyOn(request, 'getUrlTotalPosition')
        .mockImplementation(() => {
            return 'duoi'
        })

    let getData = jest.spyOn(request, 'getData')
        .mockImplementation((url) => new Promise((resolve, reject) => {
            if (url === 'tren') {
                reject('khai')
            }
            if (url === 'duoi') {
                reject('khai')
            }
        }))
    x.mappingObject = jest.fn();
    x.getDataAccount();
    setTimeout(() => {
        expect(logger).toBeCalled();
    }, 0);
})
test('getDataAccount', () => {
    x.state.accountObj = { account_id: 1 };
    let logger = jest.spyOn(log, 'error').mockImplementation(() => { })
    let geturl = jest.spyOn(request, 'getUrlAnAccount')
        .mockImplementation(() => {
            return 'tren'
        })
    let getUrlTotal = jest.spyOn(request, 'getUrlTotalPosition')
        .mockImplementation(() => {
            return 'duoi'
        })

    let getData = jest.spyOn(request, 'getData')
        .mockImplementation((url) => new Promise((resolve, reject) => {
            if (url === 'tren') {
                resolve('khai')
            }
            if (url === 'duoi') {
                resolve('khai')
            }
        }))
    x.mappingObject = jest.fn();
    x.getDataAccount();
    setTimeout(() => {
        expect(logger).toBeCalled();
    }, 0);
})
test('getDataAccount', () => {
    x.state.accountObj = { account_id: 1 };
    let geturl = jest.spyOn(request, 'getUrlAnAccount')
        .mockImplementation(() => {
            return 'tren'
        })
    let getUrlTotal = jest.spyOn(request, 'getUrlTotalPosition')
        .mockImplementation(() => {
            return 'duoi'
        })

    let getData = jest.spyOn(request, 'getData')
        .mockImplementation((url) => new Promise(resolve => {
            if (url === 'tren') {
                let datar = [{ account_name: 'khai', currency: 'usd' }]
                resolve({ data: datar })
            }
            if (url === 'duoi') {
                let datar = [{ account_name: 'khai', currency: 'usd' }]
                resolve({ data: datar })
            }
        }))
    x.mappingObject = jest.fn();
    x.getDataAccount();
    setTimeout(() => {
        expect(x.mappingObject).toBeCalled();
    }, 10);
})
test('getDataAccount', () => {
    let geturl = jest.spyOn(request, 'getUrlAnAccount')
        .mockImplementation(() => {
            return 'tren'
        })
    x.getDataAccount();
    setTimeout(() => {
        expect(geturl).not.toBeCalled();
    }, 10);
})

test('getDataAccount', () => {
    let logger = jest.spyOn(log, 'error').mockImplementation(() => { })
    x.state = null;
    x.getDataAccount();
    setTimeout(() => {
        expect(logger).toBeCalled();
    }, 0);
})

test('changeConnection', () => {
    x.refreshData = jest.fn();
    x.changeConnection(true);
    expect(x.refreshData).not.toBeCalled();
})

test('changeConnection', () => {
    x.refreshData = jest.fn();
    x.needToRefresh = true;
    x.changeConnection(true);
    expect(x.refreshData).toBeCalled();
})
