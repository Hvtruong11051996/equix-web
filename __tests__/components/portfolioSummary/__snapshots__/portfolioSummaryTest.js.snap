// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleClickIcon 1`] = `[Function]`;

exports[`render 1`] = `[Function]`;

exports[`renderInfoInline 1`] = `[Function]`;

exports[`renderInfoInline 2`] = `[Function]`;

exports[`renderInfoInline 3`] = `[Function]`;

exports[`renderInfoToggle 1`] = `[Function]`;

exports[`renderInfoTxt 1`] = `[Function]`;

exports[`renderInfoTxt 2`] = `[Function]`;

exports[`tempFormatDigit 1`] = `[Function]`;

exports[`tempFormatDigit 2`] = `[Function]`;

exports[`tempFormatDigit 3`] = `[Function]`;

exports[`tempFormatDigit 4`] = `[Function]`;

exports[`tempFormatDigitColor 1`] = `[Function]`;

exports[`tempFormatDigitColor 2`] = `[Function]`;

exports[`tempFormatDigitColor 3`] = `[Function]`;

exports[`tempFormatDigitColor 4`] = `[Function]`;

exports[`tempFormatDigitColor 5`] = `[Function]`;

exports[`tempFormatDigitColor 6`] = `[Function]`;

exports[`tempFormatDigitColor 7`] = `[Function]`;

exports[`tempFormatDigitColor 8`] = `[Function]`;

exports[`tempFormatDigitColor 9`] = `[Function]`;

exports[`tempFormatDigitColor 10`] = `[Function]`;

exports[`tempFormatDigitColor 11`] = `[Function]`;

exports[`tempFormatDigitPercent 1`] = `[Function]`;

exports[`tempFormatDigitPercent 2`] = `[Function]`;

exports[`tempFormatDigitPercent 3`] = `[Function]`;

exports[`tempFormatDigitPercent 4`] = `[Function]`;

exports[`tempFormatDigitPercent 5`] = `[Function]`;

exports[`tempFormatDigitPercent 6`] = `[Function]`;
